### PR TITLE
[Feature] Add flag to prevent remote file deletions on `theme dev`

### LINKED ISSUE #2816: [Feature]: Add --nodelete Flag to shopify theme dev
### What area(s) will this request affect?

Theme

### What type of change do you want to see?

New feature

### Overview

Add --nodelete flag to the cli command `shopify theme dev` similar like in the `shopify theme push` command

### Motivation

My use case: For project that I'm working on we are using ESBuild to handle all JS files. We used slate, but since its deprecation, there was practically no real replacement to it, so we build our own quirky and slim one. The package.json script for dev looks kinda like this:

`"start": "node esbuild.watch.js & shopify theme dev"`

and as the ESBuild is still compiling for a few seconds, shopify theme is already watching and since the asset build is not yet fully completed, deletes remotes files, to upload them again a few seconds later when the asset build is done.

### PR DESCRIPTION
**Pull request by lukeh-shopify**
### WHY are these changes introduced?

Fixes #2816 

tl;dr when bundlers are used for compilation, files are deleted/restored, which causes unecessary remote deletions and uploads

### WHAT is this pull request doing?

- Adds a new `--nodelete` flag to `theme dev`
  - prevents locally removed files from being remotely deleted on startup
  - prevents locally removed files from being remotely deleted while watching for changes

### How to test your changes?

1. Run `shopify theme dev --nodelete`
2. Delete a theme file
3. Ensure the file is not removed remotely
4. Stop the process
5. Delete another theme file
6. Run `shopify theme dev --nodelete`
7. Ensure the file is not removed remotely

### Post-release steps

[Documentation](https://shopify.dev/docs/themes/tools/cli/commands#dev) will be updated

### Measuring impact

How do we know this change was effective? Please choose one:

- [ ] n/a - this doesn't need measurement, e.g. a linting rule or a bug-fix
- [x] Existing analytics will cater for this addition
- [ ] PR includes analytics changes to measure impact

### Checklist

- [x] I've considered possible cross-platform impacts (Mac, Linux, Windows)
- [x] I've considered possible [documentation](https://shopify.dev) changes
- [x] I've made sure that any changes to `dev` or `deploy` have been reflected in the [internal flowchart](https://www.figma.com/file/7vqUp50u6dm48Zfb4JRRn8/CLI3-Internals).

### CODE REVIEW

**Review by karreiro — APPROVED** [review]
Thank you, @lukeh-shopify 🚀
